zephyr/subsys/logging/backends/log_backend_i2c.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2024 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/logging/log_backend.h>
+#include <zephyr/logging/log_core.h>
+#include <zephyr/logging/log_ctrl.h>
+#include <zephyr/logging/log_msg.h>
+#include <zephyr/logging/log_output.h>
+#include <zephyr/logging/log_backend_std.h>
+#include <zephyr/sys/__assert.h>
+#include <sedi_driver_i2c.h>
+
+static uint32_t log_format_current = CONFIG_LOG_BACKEND_I2C_OUTPUT_DEFAULT;
+static uint8_t log_i2c_buf[CONFIG_LOG_I2C_MAX_MSG_LEN];
+static bool panic_mode;
+static uint8_t i2c_bus;
+
+static int log_i2c_char_out(uint8_t *data, size_t length, void *ctx)
+{
+	ARG_UNUSED(ctx);
+
+	__ASSERT_NO_MSG(length > 0 && length <= CONFIG_LOG_I2C_MAX_MSG_LEN);
+
+	sedi_i2c_master_poll_write(i2c_bus, CONFIG_LOG_I2C_DEV_ADDR,
```

Comment:
In FCL this is OK as I2C only used as log message, but other platform shall notice we need protect

Reply:
will add protection later.